### ISSUE TITLE
Proxy to Subdirectory - Remove path from root, then add the same path back to the response from upstream server

### ISSUE THREAD

**Issue by ned-kelly** [chimurai/http-proxy-middleware#450]
Hi Guys,

So firstly apologies if this is already answered - and also hopefully I'm clearly explaining this so it makes sense....

So I'm using the middleware proxy in an express app to proxy to another microservice where i need to:

* Strip the path/sub-directory and only send to the root of the domain
* Upon the response, then add the path that was stripped back again.

The use-case is that we're actually loading an iframe to an old app that requires basic auth which we're injecting with the proxy middleware... The path (express route) to the proxy is not the root of the domain which is where things are getting mixed up....

Here's an example of the code (not working):

```javascript
    .use('/abc-proxy/', proxyMiddleware({

        target: "http://abc.internal/",
        changeOrigin: true,
        ws: true, // proxy websockets
        auth: 'username:password',

        pathRewrite: {
             '^/abc-proxy': '', // remove the base path
        },

        proxyTimeout: 60 * 1000,
        timeout: 61 * 1000,

        // Here's where it all falls apart...
        // I need to add the "/abc-proxy/" back to the response path...

        // ** This is not working **
        onProxyRes: function (proxyRes, req, res) {
            proxyRes.location = `/abc-proxy${proxyRes.location}`;
        }

    }))
```

Basically:

* Client goes to: `/abc-proxy/`
* proxyMiddleware proxies to: `http://abc.internal/` (and strips the `/abc-proxy` from the request)
* Client is then returned response from `http://abc.internal/` with the `/abc-proxy/` prefixed to the path...
* In this case it's a 301 to `http://abc.internal/some-redirect` - so I'm expecting the client to see `/abc-proxy/some-redirect`
* Instead of the above example, I'm only seeing `/some-redirect`

**Comment by ned-kelly** [comment]
Just following up here, anyone have any ideas?

**Comment by cnyet** [comment]
I don't know if I have learn your question, and I met this case that remove the base path does not working, then I edit  the property pathRewrite like this:
`pathRewrite: {'/api': '' },`
just remove ^ symbol, it works. I hop I can help you.

**detailed configuration:**
`
proxy: {
      '/api': {
        target: 'http://xxx.com',
        changeOrigin: true,
        pathRewrite: {'/api': '' },  // remove _http://xxx.com/api/abc_ to _http://xxx.com/abc_
  },
}
`

**Comment by ned-kelly** [comment]
Thanks cnyet,

Turns out the redirect was hard coded into the javascript on the client.

Will close, cheers!
